### ISSUE TITLE
libblosc compile error on ARM

### ISSUE THREAD

**Issue by bernizt** [stevengj/Blosc.jl#13]
I have tried to install the *MAT* package but unfortunately it failed due to an error when compiling one of its dependencies, *Blosc*. It looks like the make command is not using the proper arm flags and instead using the x86.
*Blosc* recently received [official ARM support](http://www.blosc.org/blog/arm-is-becoming-a-first-class-citizen-for-blosc.html) so compiling it in ARM should not be a problem.

**Julia version:** 0.4.0-dev+6443 / Commit f0328c8* / arm-linux-gnueabihf

**Julia output:**
```
Compiling libblosc...
gcc -I. -DSHUFFLE_SSE2_ENABLED -DBLOSC_SHARED_LIBRARY -fPIC -O3 -pthread -msse2 -c -o bitshuffle-sse2.o bitshuffle-sse2.c
gcc: error: unrecognized command line option ‘-msse2’
../../make.blosc:38: recipe for target 'bitshuffle-sse2.o' failed
make: *** [bitshuffle-sse2.o] Error 1
==============================================================[ ERROR: Blosc ]===============================================================

LoadError: failed process: Process(`make -f ../../make.blosc HAVE_AVX=0 LIB=../../libblosc.so`, ProcessExited(2)) [2]
while loading /root/.julia/v0.4/Blosc/deps/build.jl, in expression starting on line 21
```

**Comment by stevengj** [comment]
Is there a Julia global (analogous to `OS_NAME`) for the architecture?   If so, it would be easy to update the build script to only use the SSE stuff on x86.

**Comment by stevengj** [comment]
(The NEON instruction support does not seem to be in a Blosc release yet, but we should be able to compile a non-SIMD version in any case.)

**Comment by nalimilan** [comment]
It looks weird that the libblosc build system isn't choosing the architecture automatically, though. Maybe the next release with improved ARM support will make it work better by default?

**Comment by stevengj** [comment]
We are using our own Makefile rather than the libblosc cmake, for portability reasons (to not require cmake) and also to allow me to use the same Makefile to cross-compile the Windows DLL.

**Comment by eschnett** [comment]
This issue has been corrected on Blosc's master branch. Can you tag a new version?

**Comment by eschnett** [comment]
@stevengj Ping -- can you create a new tag for Blosc?

**Comment by eschnett** [comment]
@stevengj Blosc's master branch has interesting new functionality; could you tag a new minor version?

**Comment by tkelman** [comment]
I'd also appreciate a tag here so the package works on arm.

**Comment by stevengj** [comment]
Is blosc 1.7.1 enough to fix this?

**Comment by tkelman** [comment]
Hopefully Erik was referring to the master branch of Blosc.jl, not master of the upstream C library?

I was able to build master of Blosc.jl and it passed Pkg.test on an arm server.

**Comment by stevengj** [comment]
Ah, okay.   I want to see if I can fix #16 today, and then I'll tag a new version.

**Comment by eschnett** [comment]
Yes, Blosc master is good enough. I'm only looking for a new package tag, not a new upstream library.
